Topotato: Revamp json_cmp and add new directive: JSONCompareShouldNotExist

eznix86 opened the pull request:
fixes problem with #90 and make json_cmp more "readable"*

*need your opinion @eqvinox

eznix86 committed:
topotato: Enable every utils to be exportable

Signed-off-by: Bruno Bernard <contact.brunobernard@gmail.com>

eznix86 committed:
topotato: revamp json_cmp and add new directive

[DL: squashed:]
- fix style
- fix json compare pylint

Signed-off-by: Bruno Bernard <contact.brunobernard@gmail.com>
Signed-off-by: David Lamparter <equinox@diac24.net>

eqvinox commented:
pylint:

```
************* Module topotato.utils
topotato/utils.py:85:0: C0303: Trailing whitespace (trailing-whitespace)
topotato/utils.py:226:0: C0303: Trailing whitespace (trailing-whitespace)
topotato/utils.py:228:0: C0303: Trailing whitespace (trailing-whitespace)
topotato/utils.py:259:0: C0303: Trailing whitespace (trailing-whitespace)
topotato/utils.py:207:16: R1724: Unnecessary "else" after "continue", remove the "else" and de-indent the code inside it (no-else-continue)
topotato/utils.py:213:16: R1724: Unnecessary "else" after "continue", remove the "else" and de-indent the code inside it (no-else-continue)
topotato/utils.py:231:16: R1723: Unnecessary "elif" after "break", remove the leading "el" from "elif" (no-else-break)
topotato/utils.py:190:0: R0912: Too many branches (20/12) (too-many-branches)
topotato/utils.py:265:0: R0914: Too many local variables (17/15) (too-many-locals)
```

also needs `black` run (which would fix all the C0303)

eqvinox commented:
I'm also a bit unsure about changes in behavior tbh, there was some tricky edge cases to cover…

eznix86 commented:
fixed:
- pylint warnings

eqvinox committed:
topotato: disable leftover pylint warning

Not related to code being touched currently, just causes pre-commit hook
failure :(

Signed-off-by: David Lamparter <equinox@diac24.net>

eqvinox commented:
rebased to `topotato-base`
